dotnet/src/SemanticKernel.Skills/Skills.OpenAPI/Extensions/KernelChatGptPluginExtensions.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Resources;
+using System.Text.Json.Nodes;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.Skills.OpenAPI.Skills;
+
+namespace Microsoft.SemanticKernel.Skills.OpenAPI.Extensions;
+
+/// <summary>
+/// Class for extensions methods for IKernel interface.
+/// </summary>
+public static class KernelChatGptPluginExtensions
+{
+    /// <summary>
+    /// Imports ChatGPT plugin document from a URL.
+    /// </summary>
+    /// <param name="kernel">Semantic Kernel instance.</param>
+    /// <param name="skillName">Skill name.</param>
+    /// <param name="url">Url to in which to retrieve the ChatGPT plugin.</param>
+    /// <param name="httpClient">Optional HttpClient to use for the request.</param>
+    /// <returns>A list of all the semantic functions representing the skill.</returns>
+    public static async Task<IDictionary<string, ISKFunction>> ImportChatGptSkillFromUrlAsync(this IKernel kernel, string skillName, Uri url, HttpClient? httpClient = null)
```

Comment:
I'd stick to their terminology, "Plugin" instead of "Skill"
ImportChatGptSkill => ImportChatGptPlugin